gnd/src/main/java/com/google/android/gnd/ui/home/AddFeatureDialogFragment.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.gnd.ui.home
+
+import android.app.Dialog
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+import com.google.android.gnd.ui.common.AbstractDialogFragment
+import android.os.Bundle
+import timber.log.Timber
+import com.google.android.gnd.R
+import androidx.appcompat.app.AlertDialog
+import androidx.core.util.Consumer
+import androidx.fragment.app.FragmentManager
+import com.google.android.gnd.model.layer.Layer
+import java.lang.NullPointerException
+
+@AndroidEntryPoint
+class AddFeatureDialogFragment @Inject constructor() : AbstractDialogFragment() {
+
+    private var layerConsumer: Consumer<Layer>? = null
+    private var layers: List<Layer>? = null
```

Comment:
Should we mark these as "lazy init"? That would get rid of NPE and throw a more meaningful exception if anything is accessed before the vars are assigned. We can also cleanup the code by removing the try-catch block for NPE and !! operators.

Thoughts?

Reply:
Sorry, I meant `lateinit`. 

I think we can drop the check as well. Right now, we are failing in the else-block anyways. So, `lateinit` would take care of throwing the exception if the var is not initialized. 


```
UninitializedPropertyAccessException: lateinit property foo has not been initialized
```